scormxblock/static/js/src/scormxblock.js

```diff
@@ -125,6 +128,50 @@ function ScormXBlock_${block_id}(runtime, element) {
       return false;
     }
 
+    function disableNextModuleArrow() {
+      if (isNewUI()) {
+        disableNextModuleArrowForNewUI();
+      }
+      else {
+        disableNextModuleArrowForOldUI();
+      }
+    }
+
+    function isNewUI() {
+      return $( "body").hasClass("new-theme");
+    }
+
+    function disableNextModuleArrowForOldUI() {
+      var nextModuleLink = $( ".controls .next" );
+      if (isNextLinkAlreadyDisabled(nextModuleLink)) return;
+      nextModuleLink.addClass( "complete-scorm future" );
+      nextModuleLink.attr("href", "#");
+
+      var completionPopup = document.createElement('div');
+      completionPopup.className = "complete-scorm-content";
+      completionPopup.innerHTML = '<i class="fa fa-lock"></i>Complete all content to unlock';
+
+      $(".controls .next .mcka-tooltip").prepend(completionPopup);
+    }
+
+    function disableNextModuleArrowForNewUI() {
+      var nextModuleLink = $( ".controls .right" );
+      if (isNextLinkAlreadyDisabled(nextModuleLink)) return;
+
+      nextModuleLink.addClass( "disable" );
+      nextModuleLink.attr("href", "#");
+      var completionPopupContent = '<i class=material-icons locked>lock</i><b>Complete all content to unlock<br></b>';
+      var popupContent = $(".controls .right span").attr("data-content");
+
+      $(".controls .right span").attr("data-content", completionPopupContent + popupContent);
+    }
+
+    function isNextLinkAlreadyDisabled(nextModuleLink) {
+      var nextLinkValue = nextModuleLink.attr("href");
+      // No need to do anything since next link is already disabled
+      return nextLinkValue == "#";
```

Comment:
Instead of hash, it should be `javascript:void()`. If we will keep hash and user will click on it then the page will jump to the top.
[StackOverflow Link](https://stackoverflow.com/questions/20215248/prevent-href-link-from-changing-the-url-hash)

Reply:
Its been default practice in other cases too, e.g. in case of last module or lesson having future date.